app/lib/multiple_parameters_date_type.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class MultipleParametersDateType < ActiveModel::Type::Value
+  class PotentialDate
```

Comment:
What do you think about surfacing this class to the root namespace? I think it's a generic enough data type that it doesn't need to be namespaced by the `MultipleParametersDateType` class.